reactor-netty-core/src/main/java/reactor/netty/channel/MonoSendMany.java

```diff
@@ -578,6 +578,13 @@ public boolean trySuccess(Void result) {
 			pending--;
 
 			if (checkTerminated()) {
+				if (sourceMode == SYNC) {
+					if (requested <= REFILL_SIZE) {
+						int u = MAX_SIZE - requested;
+						requested += u;
+						nextRequest += u;
+					}
+				}
 				trySchedule();
 				return true;
 			}
```

Comment:
```suggestion
			if (checkTerminated() && sourceMode != SYNC) {
				trySchedule();
				return true;
			}
```

Can we do it like this in order to reduce duplications?